Credify/Credify/Modal/ModalViewController.swift

```diff
@@ -0,0 +1,141 @@
+//
+//  ModalViewController.swift
+//  
+//
+//  Created by Gioan Le on 28/06/2022.
+//
+
+import Foundation
+import UIKit
+
+class ModalViewController : UIViewController {
+    private var maxWidth: CGFloat {
+        get {
+            if #available(iOS 11.0, *) {
+                return view.safeAreaLayoutGuide.layoutFrame.width
+            }
+            return view.frame.width
+        }
+    }
+    
+    private var maxHeight: CGFloat {
+        get {
+            if #available(iOS 11.0, *) {
+                return view.safeAreaLayoutGuide.layoutFrame.height
+            }
+            return view.frame.height
+        }
+    }
+    
+    private var imageWidth: CGFloat {
+        get {
+            let width = 396.0
+            
+            if maxWidth < width {
+                return maxWidth
+            }
+            return width
+        }
+    }
+    
+    private var imageHeight: CGFloat {
+        get {
+            let heigh = 649.0
+            
+            if maxHeight < heigh {
+                return maxHeight
+            }
+            return heigh
+        }
+    }
+    
+    // https://i.ibb.co/bzYH2CF/offer.png
+    private let imageUrl = "https://i.ibb.co/T0sbHCC/banner-2.png"
+    
+    private var onPromotionClick: (() -> Void)!
+    
+    static func instantiate(onPromotionClick: @escaping () -> Void) -> ModalViewController {
+        let vc = ModalViewController()
+        vc.onPromotionClick = onPromotionClick
+        return vc
+    }
+    
+    override func viewDidLoad() {
```

Comment:
This function is too long. Please split some scopes into other functions.

Reply:
I updated this method